### DISCUSSION TITLE
User invitation per link, per email

### DISCUSSION THREAD

**Discussion by SzymonSel** [OrchardCMS/OrchardCore#14495] [Ideas]
Is there any work underway of creating a user invitation feature?
I out use-case we don't neccesarily want to allow any user to register but rather invite them forhand to setup their permissions etc. What we're looking for is:
* Invitation per link ie. http://demo.orchardcore.net/Register?invitationId=c4abd355aadc4d19bc854f0825ecd036
    - The invitationId would be a container for the registered users ie. this could give access to specific roles/content
* Invitation per email
    - A user account would be created for the given email, but incomplete and rather as UserId placeholder, also to grant certain permissions. All registration steps are still required - username, password, email confirmation etc.

**Comment by Piedone** [comment]
I'm not aware of any existing effort for this, but I think the two features would be basically the same. Just in the e-mail invitation, you'd include the same link. So, we're talking about and invite-by-link, and a send-link-in-email feature.

**Reply by SzymonSel** [reply]
Not quite the same.

It's like google docs sharing: 
* share by link (anyone with the link can access the doc)
* share with email (only holders of these email accounts have access)


So here we would have:
* invite by link (anyone who receive this can register, to this specific invitationId
* invite email accounts (accounts are created in advance for the specified emails) only holders of these emails may complete the registration

**Reply by Piedone** [reply]
I thought that an invite link would be specific to e-mail too, but OK then.
